controllers/numaresourcesoperator_controller.go

```diff
@@ -603,3 +590,39 @@ func validateMachineConfigLabels(mc client.Object, trees []machineconfigpools.No
 	}
 	return nil
 }
+
+func daemonsetUpdater(gdm *rtestate.GeneratedDesiredManifest) error {
```

Comment:
Maybe we can add some logs in this function (which can be enabled only when verbosity is higher) to report if we return earlier than expected in the flow.

Reply:
that's a nice idea, will add